field/track2.go

```diff
@@ -29,12 +29,41 @@ const (
 
 var track2Regex = regexp.MustCompile(`^([0-9]{1,19})(=|D)([0-9]{4})([0-9]{3})([^?]+)$`)
 
+type Track2Option func(*Track2)
+
+func WithSeparator(sep string) Track2Option {
+	return func(t *Track2) {
+		t.Separator = sep
+	}
+}
+
 func NewTrack2(spec *Spec) *Track2 {
 	return &Track2{
 		spec: spec,
 	}
 }
 
+func NewTrack2Value(
+	primaryAccountNumber string,
+	expirationDate *time.Time,
+	serviceCode,
+	discretionaryData string,
+	opts ...Track2Option,
```

Comment:
why we can't accept the separator (or format code in case of track 1) as a parameter? It looks like any other field here.

Reply:
For track2 I left `separator` as an option, because there is a default value that is valid for some networks/acquirers. So there exists situations where the caller wouldn't need to set the separator.

For track1, similar reasons. `FormatCode` and `FixedLength` are not needed for all networks/acquirers.

I think this approach makes the call site a bit cleaner:

```
Without options
fd := NewTrack1Value(
  false,
  "",
  "11111...",
  "Ben/Ross",
  "2411",
  "111",
  "1234...",
)
```

```
With options
fd := NewTrack1Value(
  "11111...",
  "Ben/Ross",
  "2411",
  "111",
  "1234...",
)

or

fd := NewTrack1Value(
  "11111...",
  "Ben/Ross",
  "2411",
  "111",
  "1234...",
  WithFixedLength(),
  WithFormatCode("X"),
)
```

Reply:
I'm not set on this approach, so if you feel strongly I can just make them params to the constructor.